### DISCUSSION TITLE
Making prefetch run at intervals

### DISCUSSION THREAD

**Discussion by tekkenking** [digitalfortress-tech/typeahead-standalone#62] [Ideas]
Thank you for the incredible work.

Here is a situation. An inventory platform; that can be accessed via mobile app and web platform simultaneously, because of this the inventory quantity can be updated by either platform. 

Prefetch got the inventory data and saved them locally (cache or local storage), then a few mins later the mobile app reduced the quantity, and the locally fetched data through Prefetch would hold the stale quantity until (I don't know when).

So If a Prefetch has the option of "interval" it would just refresh the data from the remote at a certain interval.

**Comment by niketpathak** [comment]
Hello, starting with [v4.22.0](https://github.com/digitalfortress-tech/typeahead-standalone/releases/tag/v4.22.0), there are 2 new API's available exactly for this purpose. 
Kindly checkout the API method **[reset()](https://typeahead.digitalfortress.tech/#references?id=api)** which can be used whenever you wish to flush/clear the search index. 

Optionally, you can also use the API method `addToIndex(['item1', 'item2', ...])` to populate the search index manually.

If you are using localStorage to store suggestions, then I would suggest using a library like **[localstorage-slim](https://www.npmjs.com/package/localstorage-slim)** to set a time at which data would expire from the localStorage. 

Hope that helps.
